lib/buffer.js

```diff
@@ -158,18 +157,16 @@ function assertSize(size) {
  **/
 Buffer.alloc = function(size, fill, encoding) {
   assertSize(size);
-  if (size <= 0)
-    return createBuffer(size);
-  if (fill !== undefined) {
+  if (size > 0 && fill !== undefined) {
```

Comment:
@addaleax It's not just `createUnsafeBuffer`, it's `createUnsafeBuffer` + `fill(0)`.
I believe that has been discussed before, and allocation was proven to be faster — that's why simple `Buffer.alloc(size)` does not use pool.